src/syntax_mapping.rs

```diff
@@ -122,9 +122,18 @@ impl<'a> SyntaxMapping<'a> {
             .insert("*.hook", MappingTarget::MapTo("INI"))
             .unwrap();
 
-        if let Some(xdg_config_home) = std::env::var_os("XDG_CONFIG_HOME") {
-            let git_config_path = Path::new(&xdg_config_home).join("git");
-
+        // global git config files
+        // `$XDG_CONFIG_HOME/git`, or `$HOME/.config/git` if `$XDG_CONFIG_HOME` is not set or empty
+        // see https://git-scm.com/docs/git-config#FILES
+        if let Some(git_config_path) = std::env::var_os("XDG_CONFIG_HOME")
+            .and_then(|val| (!val.is_empty()).then(|| val))
+            .map(PathBuf::from)
+            .or_else(|| std::env::var_os("HOME").map(|home| Path::new(&home).join(".config")))
```

Comment:
Thanks you for taking the time to open a PR!

I think we want to highlight `$HOME/.config/git/config` regardless of what `XDG_CONFIG_HOME` is set to though, right? I think it is perfectly reasonable to assume that the former is a git file even if the latter is set.

So it seems to me like you should move the code inside the if to a helper function and then apply it to both the possible paths to `.../git/config`.

Reply:
Yep, sounds reasonable. Will make that happen in a bit.